packages/chain-adapters/src/utxo/UtxoBaseAdapter.ts

```diff
@@ -537,8 +537,9 @@ export abstract class UtxoBaseAdapter<T extends UtxoChainId> implements IChainAd
     const { wallet, accountNumber, accountType = this.defaultUtxoAccountType } = input
 
     const bip44Params = this.getBIP44Params({ accountNumber, accountType })
-    const { xpub } = await this.getPublicKey(wallet, accountNumber, accountType)
-    const account = await this.getAccount(xpub)
+    const account = await this.getAccount(
+      input.pubKey ?? (await this.getPublicKey(wallet, accountNumber, accountType)).xpub,
```

Comment:
i would have thought the ledger hdwallet implementation would handle the concern of getting the account/pubkey, rather than the ternary in `src/context/TransactionsProvider/TransactionsProvider.tsx` which conditionally gets the account from ledger and passes in down here

Reply:
We definitely want this to be a chain-adapters concern, discussed this one with big @kaladinlight - hdwallet should return *actual* wallet data and never act as an unchained pass-through